airflow/operators/docker_operator.py

```diff
@@ -108,14 +105,47 @@ class DockerOperator(BaseOperator):
     :param shm_size: Size of ``/dev/shm`` in bytes. The size must be
         greater than 0. If omitted uses system default.
     :type shm_size: int
+    :param container_config: a dict that define all the config of the container .
+    :type container_config: dict
+        Takes a dictionary only, you can provide all the supported arguments in
+        create_container method. Check create_container method documentation for
+        more details: https://docker-py.readthedocs.io/en/latest/api.html#docker.api.
+        container.ContainerApiMixin.create_container
+        Note: the image ``image`` attribute is mandatory.
+
+        .. code-block:: python
+
+            container_config = {'image':'ubuntu:latest',
+                                'command': 'echo hello world!'
+                                'environment':{'SOME_ENV_VAR':'SOME_VALUE'},
+                                'working_dir':'/container/some_path'
+            }
+    :param host_config: a dict that define all host config of the container
+    :type host_config: dict
+        Takes a dictionary only, you can provide all the supported arguments in
+        create_host_config method. Check create_host_config method documentation for
+        more details: https://docker-py.readthedocs.io/en/latest/api.html#docker.api.
+        container.ContainerApiMixin.create_host_config
+
+        .. code-block:: python
+
+            host_config = { 'binds': ['/home/some_path:/container/some_path'],
+                            'network_mode': 'bridge',
+                            'shm_size': 1000,
+                            'cpu_shares': 2048,
+                            'mem_limit': None,
+                            'auto_remove': False,
+                            'dns': None,
+                            'dns_search': None
+            }
     """
-    template_fields = ('command', 'environment',)
+    template_fields = ('container_config', 'host_config',)
     template_ext = ('.sh', '.bash',)
 
     @apply_defaults
     def __init__(
             self,
-            image,
+            image=None,
```

Comment:
We should keep default behavior and get the value of image into `__sanitize_container_config` because if someone defined. Doing it this way will avoid variable input duplication

Reply:
On pourrait aussi garder ce `None`, et rendre ce paramètre `Deprecated`